### PR TITLE
Backend module issue

### LINKED ISSUE #13: api of Modules
add, delete, modify and select apis of Modules

### PR DESCRIPTION
**Pull request by JunkyLove**
fix #13

### CODE REVIEW

File: frontend/src/PlayBookComponents/RunBookModule.tsx [hunk 1]
```diff
@@ -30,4 +30,4 @@ function RunBookMoudle(props: moduleProp) {
   );
 }
 
-export default RunBookMoudle;
+export default RunBookModule;
```
Comment (Lingchao-Zhang) [inline frontend/src/PlayBookComponents/RunBookModule.tsx:33]: Thanks for revising my typo!

File: runbook/views.py [hunk 2]
```diff
@@ -243,31 +243,102 @@ def runbook(request, runbookId=None):
...
+    elif request.method == "POST":
+        """
+        Deal with POST request for module. Create a new module.
+
+        Args:
+            request
+        Returns:
+            return the creation result from db as JSON format.
+        """
+        try:
```
Comment (Lingchao-Zhang) [inline runbook/views.py:279]: I wonder that how new module can be saved in corresponding runbook? I think you just created a new module, but you didn't save it to the corresponding runbook. Maybe you need to get the runbook which you want to create a module, then create a new module and add this module to modules of this runbook

File: runbook/views.py [hunk 3]
```diff
@@ -243,31 +243,102 @@ def runbook(request, runbookId=None):
...
+            return the update result from db as JSON format.
+        """
+        try:
+            module_data = JSONParser().parse(request)
+            moduleId = module_data["id"]
+            try:
+                module = Module.objects.get(id=moduleId)
+                update_module(module, module_data)
+                module.editTimestamp = timezone.now
+                module.save()
```
Comment (Lingchao-Zhang) [inline runbook/views.py:308]: I think update_module function has already changed module so that module.save() may not be needed.

**Review by Lingchao-Zhang — COMMENTED** [review]
Good job for module parts in backend.
I gave some feedback on some parts.

File: runbook/views.py (hunk 2 above)
Comment (Mark-Fenng) [reply runbook/views.py:279]: Just like 'runbook''s post method, this method receives a 'runbookId' param, and then adds a new module to it.

File: runbook/views.py (hunk 3 above)
Comment (Mark-Fenng) [reply runbook/views.py:308]: I think they found a bug in my previous code. I'll fix it later.
After updating editTimestamp, we do need to save it.

File: runbook/views.py [hunk 4]
```diff
@@ -243,31 +243,102 @@ def runbook(request, runbookId=None):
...
+        Returns:
+            return the creation result from db as JSON format.
+        """
+        try:
+            module_data = JSONParser().parse(request)
+            module_data["id"] = uuid.uuid1().hex
+            new_module = Module()
+            update_module(new_module, module_data)
+            new_module.createTimestamp = timezone.now()
+            new_module.save()
```
Comment (Mark-Fenng) [inline runbook/views.py:285]:
```suggestion
```
The createTimestamp can be generated by the database automatically.

**Review by Mark-Fenng — CHANGES_REQUESTED** [review]

**Comment by Leavaway** [comment]
I've updated post method and url.py

File: runbook/serializers.py [hunk 5]
```diff
@@ -40,7 +40,7 @@ def serialize_runbook(runbook_model: Runbook, serializeDepth=1):
         "description": str(runbook_model.description),
         "instructions": str(runbook_model.instructions),
         "status": str(runbook_model.status),
-        "modules": [],
+        "modules": str([serialize_module(module) for module in runbook_model.modules.all()]),
```
[the remark this answers is not in the document]
Comment (JunkyLove) [reply runbook/serializers.py]: Thank you for your reviwing and I finally grasped the depth parameter. I have fixed it

**Review by Mark-Fenng — APPROVED** [review]

**Pull request merged**